common4j/src/main/com/microsoft/identity/common/java/nativeauth/controllers/results/ResetPasswordCommandResult.kt

```diff
@@ -46,60 +46,98 @@ interface ResetPasswordCommandResult {
         val codeLength: Int,
         val challengeTargetLabel: String,
         val challengeChannel: String,
-    ) : ResetPasswordStartCommandResult, ResetPasswordResendCodeCommandResult
+    ) : ResetPasswordStartCommandResult, ResetPasswordResendCodeCommandResult {
+        override fun toUnsanitizedString(): String = "CodeRequired(correlationId=$correlationId, codeLength=$codeLength, challengeTargetLabel=$challengeTargetLabel, challengeChannel=$challengeChannel)"
+
+        override fun toString(): String = "CodeRequired(correlationId=$correlationId, codeLength=$codeLength, challengeChannel=$challengeChannel)"
+    }
 
     data class EmailNotVerified(
+        override val correlationId: String,
         val error: String,
         val errorDescription: String,
-        override val correlationId: String
-    ) :
-        ResetPasswordStartCommandResult
+    ) : ResetPasswordStartCommandResult {
+        override fun toUnsanitizedString(): String = "EmailNotVerified(correlationId=$correlationId, error=$error, errorDescription=$errorDescription)"
+
+        override fun toString(): String = "EmailNotVerified(correlationId=$correlationId)"
+    }
 
     data class PasswordNotSet(
+        override val correlationId: String,
         val error: String,
         val errorDescription: String,
-        override val correlationId: String
-    ) :
-        ResetPasswordStartCommandResult
+    ) : ResetPasswordStartCommandResult {
+        override fun toUnsanitizedString(): String = "PasswordNotSet(correlationId=$correlationId, error=$error, errorDescription=$errorDescription)"
+
+        override fun toString(): String = "PasswordNotSet(correlationId=$correlationId)"
+    }
 
     data class UserNotFound(
+        override val correlationId: String,
         val error: String,
         val errorDescription: String,
-        override val correlationId: String
-    ) :
-        ResetPasswordStartCommandResult, ResetPasswordSubmitNewPasswordCommandResult
+    ) : ResetPasswordStartCommandResult, ResetPasswordSubmitNewPasswordCommandResult {
+        override fun toUnsanitizedString(): String = "UserNotFound(correlationId=$correlationId, error=$error, errorDescription=$errorDescription)"
+
+        override fun toString(): String = "UserNotFound(correlationId=$correlationId)"
+    }
 
     data class PasswordRequired(
         override val correlationId: String,
         val continuationToken: String
-    ) : ResetPasswordSubmitCodeCommandResult
+    ) : ResetPasswordSubmitCodeCommandResult {
+        override fun toUnsanitizedString(): String = "PasswordRequired(correlationId=$correlationId)"
+
+        override fun toString(): String = toUnsanitizedString()
+    }
```

Comment:
nit: `fun containsPii()` is not overridden here, same with `PasswordNotSet`, `UserNotFound`. The base implementation seems to have it, but probably better to explicitly state in each class.

Reply:
that's intentional; `ILoggable` defines this. I've removed all unnecessary overridings of `containsPII()`